src/index.js

```diff
@@ -0,0 +1,156 @@
+const newItem = document.querySelector('#newitem');
+const clearBtn = document.querySelector('.clr-btn');
+const localName = 'itemlist';
+
+const listSec = document.querySelector('.list');
+const submitBtn = document.getElementById('sumbit-newitem');
+const ListOfItems = new ItemList(localName);
+let dragIndex;
+
+const changeDescrip = (div, label, textArea) => {
+  label.textContent = textArea.value;
+  const id = parseInt(div.dataset.index, 10);
+  ListOfItems.updateDescrip(id, textArea.value);
+};
+
+/**
+ * Function to handle the change of description of a single item.
+ * Also updates index of all remaining items in class and data-index in DOM
+ * @param {any} div
+ * @returns {any}
+ */
+const updateDescription = (div) => {
+  const label = div.querySelector('label');
+  const textArea = div.querySelector('textarea');
+  label.style.display = 'none';
+  textArea.style.display = 'flex';
+  textArea.focus();
+  textArea.addEventListener('focusout', () => {
+    changeDescrip(div, label, textArea);
+    label.style.display = 'flex';
+    textArea.style.display = 'none';
+  });
+  textArea.addEventListener('keydown', (e) => {
+    if (e.code === 'Enter') {
+      e.preventDefault();
+      changeDescrip(div, label, textArea);
+      label.style.display = 'flex';
+      textArea.style.display = 'none';
+    }
+  });
+  textArea.addEventListener('input', () => {
+    changeDescrip(div, label, textArea);
+  });
+};
+
+/**
+ * Function to update (remove or change description) of item or items from class and DOM.
+ * Also updates index of all remaining items in class and data-index in DOM
+ * @param {any} e
+ * @returns {any}
+ */
+const updateList = (e) => {
+  e.preventDefault();
+  if (e.target.classList.contains('item-chk')) {
+    const listItem = e.target.parentNode;
+    listItem.classList.toggle('completed');
+    ListOfItems.togglecomplete(parseInt(listItem.dataset.index, 10));
+  } else if (e.target.classList.contains('icon-check')) {
+    const listItem = e.target.parentNode.parentNode;
+    listItem.classList.toggle('completed');
+    ListOfItems.togglecomplete(parseInt(listItem.dataset.index, 10));
+  } else if (e.target.classList.contains('icon-check-use')) {
+    const listItem = e.target.parentNode.parentNode.parentNode;
+    listItem.classList.toggle('completed');
+    ListOfItems.togglecomplete(parseInt(listItem.dataset.index, 10));
+  } else if ((e.target.classList.contains('app-text'))) {
+    updateDescription(e.target.parentNode.parentNode);
+  } else if (e.target.classList.contains('icon-trash-o')) {
+    const listItem = e.target.parentNode;
+    ListOfItems.removeitem(parseInt(listItem.dataset.index, 10), listItem, listUl);
+    listUl.removeChild(listItem);
+    ListOfItems.updateList('.app-item');
+  } else if (e.target.classList.contains('icon-trash-o-use')) {
+    const listItem = e.target.parentNode.parentNode;
+    ListOfItems.removeitem(parseInt(listItem.dataset.index, 10), listItem, listUl);
```

Comment:
I suggest you use a switch statement instead of if else statement to make your code clean and avoid repeatition

Reply:
Awesome catch! I made the required changes in commit 874a23333ffd8f426e141b2444dbcc9a85f6d19e